Post-Jam Improvements

OmarAlFarajat opened the pull request:
**General Improvements:**

- New input handler for scenes

- One of 5 main game BGMs are played at random (changes to the audio manager). Note, further optimizations are needed (e.g. too many find audio manager calls, TODO)

- Success and Fail screens now only change when the user ~~hits space~~ presses the "Main Menu" button, meaning they can listen to all the music

- Letterboxing package imported and added to scene cameras, so the game scales properly at different resolutions and aspect ratios!

- Fixed the bug that overloads obstacles on the screen when standing still.  

- Better aligned obstacle sprites (aligned to the bottom) 

- Better menus, halfway message, and better house controller script. The house updates more frequently now per second, so it moves smoother over the horizon. Generalized the distance/scaling instead of having hardcoded values. You can set HOUSE_TIME which is the amount of time in seconds it takes to get home with continuous walking

- Added traffic cone as a new obstacle

- Much better collision detection with obstacles (collider boxes were wrong)

- In-game pause menu added

- Added "Escape to pause game" to the main menu controls display

- Fixed audio issues related to walking

- Cleaned up some code and added comments

OmarAlFarajat committed:
Post-Jam Improvements

- New input handler for scenes

- One of 5 main game BGMs are played at random (changes to audio manager). Note, further optimizations are needed (e.g. too many find audio manager calls, TODO)

- Success and Fail screens now only change when user hits space, meaning they can listen to all the music

- Letter boxing package imported, so the game scales properly at different resolutions!

OmarAlFarajat committed:
Better aligned sprites (aligned to the bottom)

GOTLTL reviewed (changes requested):
Maybe change the volume of Easter Egg BGM to 0.3 or 0.35, it's a little too quiet right now.

and maybe tone down the volume of the footsteps by a little bit?

GOTLTL committed:
Update Omar.unity

change audio volumes

OmarAlFarajat committed:
Better Menu/Buttons, Halfway Message, Improved House Script

IrregularMeter committed:
Add Fade In/Out for halfway msg

OmarAlFarajat committed:
Improved and Added Multiple Features

- Halfway mark plays a short jingle
- Added traffic cone as new obstacle
- Much better collision detection with obstacles (collider boxes were wrong)
- In-game pause menu added
- Added "Escape to pause game" to main menu's controls panel
- Fixed audio issues related to walking
- Cleaned up some code and added comments

Attribution for halfway jingle:
https://www.zapsplat.com/music/musical-game-tone-mystery-complete-end/
